### PR TITLE
RCAL-406: Array quantities For RomanCAL

### PR DESCRIPTION
**Pull request by PaulHuwe**
Resolves [RCAL-406](https://jira.stsci.edu/browse/RCAL-nnnn)

Closes https://github.com/spacetelescope/romancal/issues/582

This PR adds support for Quantities in models required for the RomanCAL pipeline. These changes have been tested against both JWST and Roman unit tests.

**Checklist**
- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [ ] updated relevant tests
- [ ] updated relevant documentation
- [x] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

**Comment by PaulHuwe** [comment]
This passes all unit tests throughout the RomanCAL pipeline and the unit tests in jwst.

**Review by ddavis-stsci — APPROVED** [review]
LGTM

File: src/stcal/dark_current/dark_class.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import numpy as np
+from astropy import units as u
```
Comment (kmacdonald-stsci) [inline src/stcal/dark_current/dark_class.py:2]: Is this a standard import idiom for astropy units?  If not, using just 'u' may be too terse.

**Review by kmacdonald-stsci — APPROVED** [review]

File: src/stcal/dark_current/dark_class.py (hunk 1 above)
Comment (PaulHuwe) [reply src/stcal/dark_current/dark_class.py:2]: It is: https://docs.astropy.org/en/stable/units/index.html

**Review by schlafly — APPROVED** [review]
Looks good to me.  One thing to think about: do we really want these comparisons to be type(x) == u.Quantity?  I usually use isinstance(x, u.Quantity) in these cases to catch subclasses (though I'm not sure if any such subclasses currently exist).  I could also doing something more duck-type-y along the lines of:
```
try:
    x.data = model.data.value
except AttributeError:
   x.data = model.data
```
which would be more flexible for things implementing quantity-like interfaces but could have unintended consequences.

Anyway, some thoughts, but looks good to me to merge.

**Comment by PaulHuwe** [comment]
> Looks good to me. One thing to think about: do we really want these comparisons to be type(x) == u.Quantity? I usually use isinstance(x, u.Quantity) in these cases to catch subclasses (though I'm not sure if any such subclasses currently exist). I could also doing something more duck-type-y along the lines of:
> 
> ```
> try:
>     x.data = model.data.value
> except AttributeError:
>    x.data = model.data
> ```
> 
> which would be more flexible for things implementing quantity-like interfaces but could have unintended consequences.
> 
> Anyway, some thoughts, but looks good to me to merge.

I like this and I will implement the change.

**Comment by PaulHuwe** [comment]
I suspect I lack merge permissions for STCAL - could someone merge this?

**Pull request merged**